### DISCUSSION TITLE
State of the plugin

### DISCUSSION THREAD

**Discussion by thomassth** [builttoroam/device_calendar#339] [General]
Currently, all 3 of the maintainers are not very active to this plugin (yes, including me)

We have committed to publish a new version of the plugin, with time zone support, null safety and other fixes in mind, but we have yet to complete this. Documentations, self-tests and other details also needed some work.

Even if I spend more time on this project, a number of functions are still restricted, like I can’t review PRs done by me, and because other maintainers are not available, I basically cannot commit anything. Discussions were also not enabled here, hence I’m opening an issue instead. I’m also not sure if I have the permission push the build to prerelease or release channel of pub.dev

Perhaps new maintainers may help, although that would also need @nickrandolph ’s approval.

If you’re interested or consider this plugin useful, feel free to chip in.

**Comment by nickrandolph** [comment]
@thomassth happy to add new maintainers if anyone is interested
Discussions are now enabled - thanks for raising this as I wasn't aware it wasn't enabled by default
With regards to PRs, please feel free to ping me on Twitter (@thenickrandolph) if you need a PR reviewed so it can be merged. I'm not currently spending much time on this project but happy to jump on and grease the wheels for any PR that's blocked waiting for review.
With regards to pushing new versions, this is automated - you should be able to push a PR to master branch in order to trigger an actual release.

**Comment by thomassth** [comment]
Thank you @nickrandolph for the reply

If anyone wish to be a maintainer of this package feel free to ask. 

Mostly you need to review and approve PRs and issues. And preferably an android or iOS device on hand to run and test things. 

It would be better if you have some experience writing flutter, kotlin, swift or flutter plug-ins, but that's not a must.

**Reply by sowens-csd** [reply]
I'll put my name in as possible maintainer if you need someone. I use and like the plugin so I'm happy to help if I can. I have built and shipped other Flutter plugins for iOS and Android. 

On the con side I hate date/time issues with the burning passion of a thousand suns going supernova, but that probably only puts me in the top 97% of developers.

**Comment by nickrandolph** [comment]
@thomassth what do you feel are the "must-dos" before we push a new version of the plugin?

**Reply by thomassth** [reply]
I think we should be fixing current bugs like #336 , and have a proper migration plan from previous versions due to the new time zone settings (see #308). Having a bull-safe package that runs properly  should be the first thing for now.

Other features can wait for .1 release.

**Comment by thomassth** [comment]
@jifisherki @IVLIVS-III @kryjomskiDev Seeing your recent comments and PRs, I think you would be a suitable maintainer of this plugin.

Among other things, you'll be able to review PRs, and push to pre-release / release of pub.dev

If you're interested, let @nickrandolph know. Think he's reachable on Twitter DMs.

If anyone else is interested, let me know.

**Reply by nickrandolph** [reply]
If you are interested in becoming a maintainer, just reply here and I can add you

**Reply by IVLIVS-III** [reply]
hey, thanks for the offer.
Unfortunately I feel, that I have neither the time nor other resources to be comfortable with taking the responsibility that comes with being a maintainer out of the box.

I do appreciate (und use) this plugin, and whenever possible I try to help out with a PR.

Also I don't really have any previous experience with the non-Dart i.e. platform-specific side of plugins.
Although I do have multiple years of experience regarding Flutter and Dart.

If, however, as little as reviewing PRs and finalizing new versions once or twice a month is enough to share the burden of maintaining this plugin, I'm willing to help out and accept the responsibility that comes with joining the maintainer-team.

Please note, that maintaining this plugin would not be one of my highest priorities, tho.

It would truly be sad to see this plugin die caused solely by a lack of multiple minor contributions.

In case you have any questions regarding this message, feel free to reply.

And if it is helpful for the plugin and you are comfortable with adding me as maintainer despite my commitment only to little work, I'm happy to help out.

**Reply by thomassth** [reply]
just coming here to look around from time to time would have already helped
tbh none of us did this as the highest priority, we just chip in our free time when we can
such is the fate of one of the essential plugins of a quickly emerging platform, with one of the biggest company pushing it

**Reply by IVLIVS-III** [reply]
In that case, I'm happy to join as maintainer (or rather "Collaborator" to use the correct Github term ;)

**Reply by thomassth** [reply]
@nickrandolph can you help us out?

**Reply by nickrandolph** [reply]
Hey sorry let me jump on this evening

**Reply by nickrandolph** [reply]
@IVLIVS-III sorry about the delay - you should have got an invite (apologies if you got multiple invites - I had to cancel the first couple as they were from the wrong team). Let me know if any issues.
